packages/database/src/api/Reference.ts

```diff
@@ -570,95 +558,61 @@ export class Reference extends Query implements Compat<ReferenceImpl> {
    *
    * Externally - this is the firebase.database.Reference type.
    */
-  constructor(database: Database, path: Path) {
+  constructor(readonly database: Database, readonly _delegate: ReferenceImpl) {
     super(
       database,
-      new QueryImpl(database._delegate._repo, path, new QueryParams(), false)
+      new QueryImpl(_delegate._repo, _delegate._path, new QueryParams(), false)
     );
   }
 
   /** @return {?string} */
   getKey(): string | null {
     validateArgCount('Reference.key', 0, 0, arguments.length);
-
-    if (pathIsEmpty(this._delegate._path)) {
-      return null;
-    } else {
-      return pathGetBack(this._delegate._path);
-    }
+    return this._delegate.key;
   }
 
-  child(pathString: string | Path): Reference {
+  child(pathString: string): Reference {
```

Comment:
Is it a breaking change?

Reply:
Path is an internal class, so it should be fine.